src/core/structures/PDFAcroFormField.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  PDFArray,
+  PDFDict,
+  PDFName,
+  PDFNumber,
+  PDFObject,
+  PDFString,
+} from 'src/core';
+
+const acroFormFieldTypes: Array<PDFObject | undefined> = [
+  PDFName.Btn,
+  PDFName.Ch,
+  PDFName.Tx,
+  PDFName.Sig,
+];
+
+class PDFAcroFormField {
+  static fromDict(dict: PDFDict): PDFAcroFormField {
+    const ft = PDFName.of('FT');
+    const hasValidFieldType =
+      dict.has(ft) && acroFormFieldTypes.includes(dict.get(ft));
+    if (!hasValidFieldType) {
+      throw new Error('Invalid PDFAcroFormField Type');
+    }
+    return new PDFAcroFormField(dict);
+  }
+
+  private readonly dict: PDFDict;
+
+  protected constructor(dict: PDFDict) {
+    this.dict = dict;
+  }
+
+  FT(): PDFName {
```

Comment:
This field is actually only required for terminal fields. AcroFields can actually be structured as a tree. In which case, the branch nodes will not have a `FT`.

Speaking of which, it might be useful to distinguish between terminal an non-terminal fields. Because the `Kids` field is required for non-terminal fields, but is disallowed on terminal ones. We could structure it like so:

```js
class PDFAcroField // "dummy" base class, similar to `PDFObject`
class PDFTerminalField extends PDFAcroField
class PDFNonTerminalField extends PDFAcroField
```

But I'm open to alternatives as well. Or perhaps we should wait to build this abstraction until more work has been done on some of the other AcroForm API cards.

Reply:
Updated